src/Symfony/Component/Security/Core/Authorization/Voter/AbstractVoter.php

```diff
@@ -65,6 +65,12 @@ public function vote(TokenInterface $token, $object, array $attributes)
         // abstain vote by default in case none of the attributes are supported
         $vote = self::ACCESS_ABSTAIN;
 
+        $reflector = new \ReflectionMethod($this, 'voteOnAttribute');
+        $isNewOverwritten = $reflector->getDeclaringClass()->getName() !== 'Symfony\Component\Security\Core\Authorization\Voter\AbstractVoter';
+        if (!$isNewOverwritten) {
+            @trigger_error(sprintf("The AbstractVoter::isGranted method is deprecated since 2.8 and won't be called anymore in 3.0. Override voteOnAttribute() instead.", $reflector->class), E_USER_DEPRECATED);
```

Comment:
you should mention the class of ``$this`` so that people can know in which class the change is needed.